src/main/flight/imu.c

```diff
@@ -807,7 +972,12 @@ void imuSetHasNewData(uint32_t dt)
 bool imuQuaternionHeadfreeOffsetSet(void)
 {
     if ((abs(attitude.values.roll) < 450)  && (abs(attitude.values.pitch) < 450)) {
-        const float yaw = -atan2_approx((+2.0f * (qP.wz + qP.xy)), (+1.0f - 2.0f * (qP.yy + qP.zz)));
+        const quaternion* q = &ahrsState.nominal.attitude;
```

Comment:
(this should be considerably simpler when using  swing-twist  decomposition to Z axis. But see above :-( )